### ISSUE TITLE
Intellisense does not appear off of generic type

### ISSUE THREAD

**Issue by KevinRansom** [Microsoft/visualfsharp#8]
Opened by: [latkin](https://www.codeplex.com/site/users/view/latkin)

In the below code, intellisense no longer pops up at the specified point. It used to work ok in 3.1.X, but has stopped working in 4.0
````fsharp
type GT<'a> =
  static member P = 12
  static member Q = 13

let _ = GT<int>. <--- no intellisense here now, expecting list with 'P','Q'
let _ = GT.   <---- intellisense does still appear here
````
Testing it out on some older builds, this appears to have been introduced with the "constructors as first-class functions" feature.

>comments
>>dsyme wrote Dec 11, 2014 at 1:35 PM [x] 

>>What's the name of the specific IDE unit test that's failing? I'll aim to run them and make the fix tomorrow 

>>>latkin wrote Dec 11, 2014 at 2:12 PM [x] 
>>>UnitTests.Tests.LanguageService.AutoCompletionListTests, testcase Generics.GenericTypeMembers

>>>>latkin wrote Dec 11, 2014 at 2:30 PM [x] 
>>>>Note that you'll need a build of VS 2015, plus the VS 2015 SDK, installed to run the IDE tests for fsharp4. You can get both from here  http://www.visualstudio.com/en-us/downloads/visual-studio-2015-downloads-vs.aspx

**Comment by latkin** [comment]
@dsyme this regressed (I'm pretty sure) as part of the "ctors as first-class functions" work.

**Comment by dsyme** [comment]
Have assigned it to myself

**Comment by KevinRansom** [comment]
Don,
is there any progress on this?  It is a regression and so it is one we would prefer to have a fix for when we release F# 4.0

**Comment by dsyme** [comment]
I've looked at it several times and not found the mistake.  I'll keep looking :)

**Comment by dsyme** [comment]
I made my third attempt on this and finally got somewhere.  It was hard!  At least I now understand what changed between F# 3.1 and F# 4.0 and have prepared a potential fix. https://github.com/Microsoft/visualfsharp/pull/354

I'd appreciate help in testing if the fix works "for real" in intellisense in VS2015. I've tried to make the fix as non-intrusive and low-risk as possible (even though that involved duplicating some code).
